politeiawww/user/cockroachdb/cockroachdb_test.go

```diff
@@ -0,0 +1,362 @@
+// +build cockroachdb
+
+// The tests in this file need cockroachdb to be up and will only be run if you
+// specify "-tags=cockroachdb" with `go test`.
+
+package cockroachdb
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/decred/politeia/politeiawww/user"
+	"github.com/google/uuid"
+
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+)
+
+func modelToSession(s Session) user.Session {
+	return user.Session{
+		ID:     s.ID,
+		UserID: s.UserID,
+		Values: s.Values,
+	}
+}
+
+var testDBConnection *cockroachdb
+
+func connectToTestDB(t *testing.T) *cockroachdb {
+	home := os.Getenv("HOME")
+	var (
```

Comment:
These variables really shouldn't be hard coded.  Would it make sense here to load the config file?  

Ideally, this setup would only be done once.  Instead of running them all in a single test like you were doing before, would using a `TestMain` make sense here?

Reply:
I will look into a way to tidy up the variables. Will also look at `TestMain` -- having said that the setup only really happens for the very first test as you can see here: https://github.com/al-maisan/politeia/blob/bece742a6321ec7de3a5c79da65c5d4951717a50/politeiawww/user/cockroachdb/cockroachdb_test.go#L42

Reply:
Alright, the setup part isn't that big of a deal.  I'm more concerned about the hard coded file paths since they're config params and might be different on various machines.

Reply:
Also, check your riot DMs.

Reply:
I am travelling and have no access to riot -- sorry

Reply:
BTW, this should be resolved as well. I have moved the config to a separate file.

Reply:
https://github.com/al-maisan/politeia/blob/647-pwd-change/politeiawww/user/cockroachdb/testdata/test.conf

Reply:
Hey, it would be nice to wrap this up eventually .. can you please let me know where things stand? Thank you.

Reply:
I haven't forgotten about this.  I'll take another look at it today.

Reply:
Wouldn't it be simpler to pull these config values from the politeiawww.conf file instead of creating a whole new config file?

Reply:
Well, for starters the test configuration may be different from the one in `politeiawww.conf`. Furthermore 2 of the config items (`network` and `encryptionkey`) aren't even contained in the latter.

Reply:
Also, how would I know where to find the `politeiawww.conf` file -- as you mentioned in the original comment it may be in different locations on various machines.

Reply:
> Furthermore 2 of the config items (network and encryptionkey) aren't even contained in the latter.

They are.
https://github.com/decred/politeia/blob/master/politeiawww/config.go#L99
https://github.com/decred/politeia/blob/master/politeiawww/config.go#L125

> Also, how would I know where to find the politeiawww.conf file

The politeiawww.conf file is required to be at the path `appDataDir/politeiawww.conf`.  `appDataDir` will be different for the various operating systems.  There is a function to retrieve the `appDataDir` that takes into account the operating system.

tbh I'd prefer we just remove these cockroachdb tests at this point.  If we're going to be running tests against a cockroachdb instance we need to write some testing infrastructure that uses a separate test database with proper setup and teardown and that pulls in the config settings from the politiawww.conf file.  That's out of the scope of this PR though.  I don't have a problem with skipping it and getting this PR merged.

Reply:
cockroachdb tests removed